### PR TITLE
Small and incremental refactorings

### PR DESCRIPTION
**Pull request by ZaqueuCavalcante**
I will try to make surgical changes now. I will not change the name of any class, method or package. Only attributes and implementations. In this way, no interface will be broken.

### CODE REVIEW

File: src/main/java/technology/tabula/Page.java [hunk 1]
```diff
@@ -13,243 +13,271 @@
 // TODO: this class should probably be called "PageArea" or something like that
 public class Page extends Rectangle {
```
Comment (ZaqueuCavalcante) [inline src/main/java/technology/tabula/Page.java:18]: I realized now that revising this change is going to be difficult. If you open both files you will notice that I just moved the methods.

File: src/main/java/technology/tabula/CohenSutherlandClipping.java [hunk 2]
```diff
@@ -34,100 +34,108 @@
...
+
+        Point(double x, double y) {
+            setPositionAndRegion(x, y);
+        }
+
+        void setPositionAndRegion(double x, double y) {
+            this.x = x; this.y = y;
+            region = (x < xMin) ? LEFT : (x > xMax) ? RIGHT : INSIDE;
+            if (y < yMin)
+                region |= BOTTOM;
```
Comment (ZaqueuCavalcante) [inline src/main/java/technology/tabula/CohenSutherlandClipping.java:131]: I think that in this part the operations are reversed. The logical OR should be done with the constant TOP. See [here](https://en.wikipedia.org/wiki/Cohen%E2%80%93Sutherland_algorithm) the bits table.

File: src/main/java/technology/tabula/CohenSutherlandClipping.java [hunk 3]
```diff
@@ -74,25 +74,25 @@ public boolean clip(Line2D.Float line) {
         double lineSlope = lineIsVertical ? 0d : (point2.y-point1.y)/(point2.x-point1.x);
 
         while (point1.region != INSIDE || point2.region != INSIDE) {
-            if ((point1.region & point2.region) != INSIDE) return false;
+            if ((point1.region & point2.region) != 0) return false;
```
Comment (ZaqueuCavalcante) [inline src/main/java/technology/tabula/CohenSutherlandClipping.java:77]: Here (and in other altered Ifs) are comparing the result of the logical OR with zero.

File: src/main/java/technology/tabula/CohenSutherlandClipping.java [hunk 4]
```diff
@@ -74,25 +74,25 @@ public boolean clip(Line2D.Float line) {
         double lineSlope = lineIsVertical ? 0d : (point2.y-point1.y)/(point2.x-point1.x);
 
         while (point1.region != INSIDE || point2.region != INSIDE) {
```
Comment (ZaqueuCavalcante) [inline src/main/java/technology/tabula/CohenSutherlandClipping.java:76]: Here we are comparing point region with other region.

File: src/main/java/technology/tabula/CohenSutherlandClipping.java [hunk 5]
```diff
@@ -74,25 +74,25 @@ public boolean clip(Line2D.Float line) {
...
-            if ((point1.region & point2.region) != INSIDE) return false;
+            if ((point1.region & point2.region) != 0) return false;
 
             outsidePoint.region = (point1.region == INSIDE) ? point2.region : point1.region;
```
Comment (ZaqueuCavalcante) [inline src/main/java/technology/tabula/CohenSutherlandClipping.java:79]: Here we are comparing point region with other region.

**Comment by ZaqueuCavalcante** [comment]
I think it's good for that pull request. I'm waiting to your review right now, @jazzido.

**Review by jazzido — APPROVED** [review]
This is really impressive @ZaqueuCavalcante! On behalf of the Tabula team, let me thank you for such a thorough read and clean up of our code base.

I'm approving and merging your changes to master. If there are other changes that you'd like to work on, please get in touch to my email address.

**Pull request merged**